src/inference/src/cpp_interfaces/interface/ie_iinfer_request_internal.cpp

```diff
@@ -157,6 +158,123 @@ void IInferRequestInternal::SetBlob(const std::string& name, const Blob::Ptr& us
+        SetBlob(name, blobs[0]);
+        return;
+    }
+    std::shared_ptr<const ov::op::v0::Parameter> param;
+    const auto& inputs = GetInputs();
+    for (const auto& input : inputs) {
+        if (auto p = std::dynamic_pointer_cast<const ov::op::v0::Parameter>(input)) {
+            const auto& names = p->output(0).get_tensor().get_names();
+            if (names.find(name) != names.end()) {
+                param = p;
+                break;
+            }
+        }
+    }
+    OPENVINO_ASSERT(param, "set_input_tensors/set_tensors error. Parameter '", name, "' is not found");
+    OPENVINO_ASSERT(ov::layout::has_batch(param->get_layout()),
+                    "set_input_tensors/set_tensors can be used only for inputs with N(batch) dimension"
+                    " 'layout' defined. Current layout for '",
+                    name,
+                    "' is ",
+                    param->get_layout().to_string());
+    auto batch_idx = ov::layout::batch_idx(param->get_layout());
+    if (batch_idx < 0) {
+        batch_idx += static_cast<int64_t>(blobs[0]->getTensorDesc().getDims().size());
+    }
+    auto input_size = std::accumulate(blobs.begin(), blobs.end(), 0, [&batch_idx](int res, const Blob::Ptr& item) {
+        OPENVINO_ASSERT(item->getTensorDesc().getDims()[batch_idx] == 1,
+                        "set_input_tensors/set_tensors. Tensors shall represent one item in a batch, ",
+                        item->getTensorDesc().getDims()[batch_idx],
+                        " provided");
+        return res + item->getTensorDesc().getDims()[batch_idx];
+    });
+    if (param->get_partial_shape().rank().is_static()) {
+        OPENVINO_ASSERT(batch_idx >= 0 && batch_idx < param->get_partial_shape().rank().get_length(),
+                        "set_input_tensors/set_tensors error. Layout ",
+                        param->get_layout().to_string(),
+                        " is incorrect for input '",
+                        name,
+                        "' with shape ",
+                        param->get_partial_shape());
+        auto batch = param->get_partial_shape()[batch_idx];
+
+        OPENVINO_ASSERT(batch.is_dynamic() || batch.get_length() == input_size,
+                        "set_input_tensors/set_tensors error. Input shape ",
+                        param->get_partial_shape(),
+                        "batch ",
+                        batch,
+                        "doesn't match with total blobs count: ",
+                        input_size);
+    }
+    OPENVINO_ASSERT(batch_idx == 0,
+                    "set_input_tensors/set_tensors is not currently supported for batch dimension index ",
+                    batch_idx,
+                    " != 0");
+    // Initial implementation always performs creation of new Blob and 'memcpy'
+    // In future consider checking if blobs point to contiguous range of memory
+    auto& desc = blobs[0]->getTensorDesc();
+    desc.getDims()[batch_idx] = input_size;
```

Comment:
why do we patch original blob tensor descriptor? Can we just take dims and patch batch dimension?